### PR TITLE
fix(ci): separate rpm/bin/bin-static/deb packages before publication, rename bin-static

### PR DESCRIPTION
**Pull request by LucaGuerra**
**What type of PR is this?**

/kind bug
/area CI

**What this PR does / why we need it**:

This fixes a couple things with the publish package pipeline
* Now different types of packages are extracted to different directories. In the current state, static overwrites bin!
* The static artifact is renamed before being uploaded to maintain consistency with the artifact names

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: .github/workflows/reusable_publish_packages.yaml [hunk 1]
```diff
@@ -47,31 +47,31 @@ jobs:
         uses: actions/download-artifact@v3
         with:
           name: falco-${{ inputs.version }}-x86_64.rpm
-          path: /tmp
+          path: /tmp/falco-rpm
```
Comment (LucaGuerra) [inline .github/workflows/reusable_publish_packages.yaml:50]: according to the documentation ( https://github.com/actions/download-artifact#compatibility-between-v1-and-v2v3 ) the directory should be created if it doesn't exist.

**Review by Andreagit97 — APPROVED** [review]
/approve

**Comment by FedeDP** [comment]
Refs #2501

**Pull request merged**